Dockerfile

```
@@ -52,5 +52,35 @@ COPY --from=nim-build /app/migrations/ /app/migrations/
 RUN ln -sv /usr/local/bin/$MAKE_TARGET /usr/bin/wakunode
 
 ENTRYPOINT ["/usr/bin/wakunode"]
+
 # By default just show help if called without arguments
 CMD ["--help"]
+
+
+# DEBUG IMAGE ------------------------------------------------------------------
+
+# Build debug tools: heaptrack
+FROM alpine:edge AS heaptrack-build
+
+RUN apk update
+RUN apk add -- gdb git g++ make cmake zlib-dev boost-dev libunwind-dev
+RUN git clone https://github.com/KDE/heaptrack.git /heaptrack
+
+WORKDIR /heaptrack/build
+RUN cmake -DCMAKE_BUILD_TYPE=Release ..
+RUN make -j$(nproc)
+
+
+# Debug image
+FROM prod AS debug
+
+RUN apk add --no-cache gdb
```

It would be good to only call apk once

```suggestion
RUN apk add --no-cache gdb libunwind
```

---

Good point! I'll do that thanks !